qiskit_nature/operators/second_quantization/qubit_converter.py

```diff
@@ -11,26 +11,92 @@
 from qiskit_nature.mappers.second_quantization import QubitMapper
 
 from . import SecondQuantizedOp
 
+logger = logging.getLogger(__name__)
+
 
 class QubitConverter:
     """A converter from Second-Quantized to Qubit Operators."""
 
     def __init__(self,
-                 mappers: Union[QubitMapper, Tuple[QubitMapper, QubitMapper]],
+                 mapper: QubitMapper,
+                 two_qubit_reduction: bool = False,
                  z2symmetry_reduction: Optional[List[int]] = None):
 
-        self._mappers = mappers
-        self._z2symmetry_reduction = z2symmetry_reduction
-        self._z2symmetries = None
+        self._mapper = mapper
+        self._two_qubit_reduction = two_qubit_reduction
+        self._z2symmetry_reduction = None
+        self.z2symmetry_reduction = z2symmetry_reduction  # Setter does validation
+
+        self._conversion_done = False
+        self._did_two_qubit_reduction = False
+        self._num_particles = None
+        self._z2symmetries = self._no_symmetries
+
+    def _invalidate(self):
+        self._conversion_done = False
+        self._did_two_qubit_reduction = False
+        self._num_particles = None
+        self._z2symmetries = self._no_symmetries
+
+    def _set_valid(self, did_two_qubit_reduction: bool,
+                   num_particles: Optional[Tuple[int, int]],
+                   z2symmetries: Z2Symmetries):
+        self._did_two_qubit_reduction = did_two_qubit_reduction
+        self._num_particles = num_particles
+        self._z2symmetries = z2symmetries
+        self._conversion_done = True
+
+    def _check_valid(self, converting: bool = False):
+        if not self._conversion_done:
+            if not converting:
+                raise QiskitNatureError("Properties are valid only after a convert() has "
+                                        "been called successfully. "
+                                        "Conversion state is reset if any settings are updated.")
+            else:
+                raise QiskitNatureError("convert_more() can only be used after a convert() has "
+                                        "been called successfully as long as no settings "
+                                        "were updated since that call.")
+
+    @property
+    def _no_symmetries(self) -> Z2Symmetries:
+        return Z2Symmetries([], [], [], None)
+
+    @property
+    def mapper(self) -> QubitMapper:
+        """Get mapper"""
+        return self._mapper
+
+    @mapper.setter
+    def mapper(self, value: QubitMapper) -> None:
+        """Set mapper"""
+        self._mapper = value
+        self._invalidate()
+
+    @property
+    def two_qubit_reduction(self) -> bool:
+        """Get two_qubit_reduction"""
+        return self._two_qubit_reduction
+
+    @two_qubit_reduction.setter
+    def two_qubit_reduction(self, value: bool) -> None:
+        """Set two_qubit_reduction"""
+        self._two_qubit_reduction = value
+        self._invalidate()
```

Comment:
Do we always want to invalidate or only when this value actually changed?
The same can be asked for the other setters.

Reply:
I did not make any value check - it seemed to me an unnecessary complication. If you were setting it you were likely going to reuse it for a different conversion rather than do more of the same conversions. Yes you might set it exactly the same but as I said I did not believe it was worth the extra value checks